src/main/java/practice/CandidateValidator.java

```diff
@@ -1,5 +1,43 @@
 package practice;
 
-public class CandidateValidator {
-    //write your code here
+import java.util.function.Predicate;
+import model.Candidate;
+
+public class CandidateValidator implements Predicate<Candidate> {
+    private final String slashSeparator = "-";
+
+    @Override
+    public boolean test(Candidate candidate) {
+        if (validCandidateAge(candidate) && validNationality(candidate)
+                && validPeriodInUkr(candidate) && candidate.isAllowedToVote()) {
+            return true;
+        }
+        return false;
```

Comment:
You can simplify this
```suggestion
        return validCandidateAge(candidate) 
        && validNationality(candidate)
        && validPeriodInUkr(candidate) 
        && candidate.isAllowedToVote()

```